electrum/lnworker.py

```diff
@@ -2717,7 +2717,13 @@ def send_capacity(chan):
                     delta_msat = 0
             else:
                 delta_msat = 0
-            return chan.available_to_spend(LOCAL) + delta_msat
+            if single_payment:
+                try:
+                    return chan.determine_payment_constraints(LOCAL).htlc_maximum_msat + delta_msat
+                except PaymentFailure:
+                    return 0
+            else:
+                return chan.available_to_spend(LOCAL) + delta_msat
```

Comment:
why not pass single_payment to `chan.available_to_spend`, and let that method do the if branching?

That way, you could write a unit test directly in test_lnchannel. Currently, a unit test would need to be added to test_lnpeer. However, the property we are testing does not really belong in lnpeer, it is lower level.

Reply:
Done in e83055c4339024ac52e4cf02a9cedef577632906

Reply:
ok, I see you have removed `determine_payment_constraints`, that looks good. However, in `_assert_can_add_htlc`, you are calling `available_to_spend` without passing `single_payment=True.` I guess that is oversight?

You should really add a unit test in test_lnchannel, that tests adding an htlc above max_inflight. It would have detected that.

Reply:
Yeah, sorry, I should've put this to draft while working on this, it was not finished yet :)

Reply:
> I can remove that part and let available_to_spend handle this, but then we lose the specific error message.

I created helper functions to avoid code duplication, pushed in your repo